azavea/cac-tripplanner
# Feature/kak/multiple destination images#1014

## Overview

Add extra images fields to destinations and events. Adds cropping widget to admin interface and removes aspect ratio restriction on image file uploads, which is no longer needed, as the images are cropped instead. The extra image files are cropped to both the wide and narrow image sizes.

Displays extra images, if any, in a carousel with the primary wide image for a destination or event, if present. If destination or event has no extra images, continues to display single wide image still.


### Demo

Events carousel:
![output](https://user-images.githubusercontent.com/960264/38751842-458f90e0-3f27-11e8-80b0-094365aec656.gif)

Destinations required images with cropping tool:
![image](https://user-images.githubusercontent.com/960264/38751906-84c3382a-3f27-11e8-9f3a-59b7a5dc51db.png)

Inline admin extra images:
![image](https://user-images.githubusercontent.com/960264/38751982-b7e7844a-3f27-11e8-9a9f-249ca1cf1871.png)


### Notes

It's necessary to save the admin form after uploading an image file and go back into the form to see the cropping widget for the new image.


## Testing Instructions

 * Reprovision app server: `vagrant provision app`
 * Cropping images should work as expected for articles, destinations, and events
 * Uploading extra images should work as expected
 * Detail views for events and destinations should display extra images in a resizable carousel
 * Check `extra_images` and `extra_wide_images` [JSON](http://localhost:8024/api/destinations/search?text=)

Closes #1014
Closes #959 
Connects #1021
